discovery-provider/src/queries/search_queries.py

```diff
@@ -450,59 +432,76 @@ def track_search_query(
     res = sqlalchemy.text(
         # pylint: disable=C0301
         f"""
-        select track_id from (
-            select
-                track_id,
-                (sum(score) + (:title_weight * similarity(coalesce(title, ''), query))) as total_score
-            from (
-                select
-                    d."track_id" as track_id, d."word" as word, similarity(d."word", :query) as score,
-                    d."track_title" as title, :query as query
-                from "track_lexeme_dict" d
-                {
-                    'inner join "saves" s on s.save_item_id = d.track_id'
-                    if personalized and current_user_id
-                    else ""
-                }
-                {
-                    'inner join "tracks" t on t.track_id = d.track_id'
-                    if only_downloadable
-                    else ""
-                }
-                where d."word" % :query
-                {
-                    "and s.save_type='track' and s.is_current=true and " +
-                    "s.is_delete=false and s.user_id = :current_user_id"
-                    if personalized and current_user_id
-                    else ""
-                }
-                {
-                    "and (t.download->>'is_downloadable')::boolean is True"
-                    if only_downloadable
-                    else ""
-                }
-            ) as results
-            group by track_id, title, query
-        ) as results2
-        order by total_score desc, track_id asc
+        select track_id, b.balance, b.associated_wallets_balance from (
```

Comment:
I think we should probs explore what the query planner is doing a bit more in some of these queries to find if there's anything else worth optimizing. structurally this makes a lot of sense--but maybe something else we can optimize?

Reply:
discussed offline - perf is equivilent w gzip